plugins/main/bridge/bridge.go

```diff
@@ -485,41 +493,44 @@ func cmdDel(args *skel.CmdArgs) error {
 		return err
 	}
 
-	if err := ipam.ExecDel(n.IPAM.Type, args.StdinData); err != nil {
-		return err
-	}
-
 	if args.Netns == "" {
 		return nil
 	}
 
-	// There is a netns so try to clean up. Delete can be called multiple times
-	// so don't return an error if the device is already removed.
-	// If the device isn't there then don't try to clean up IP masq either.
-	var ipnets []*net.IPNet
-	err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
-		var err error
-		ipnets, err = ip.DelLinkByNameAddr(args.IfName)
-		if err != nil && err == ip.ErrLinkNotFound {
-			return nil
+	isLayer3 := n.IPAM.Type != ""
+
+	if isLayer3 {
+		if err := ipam.ExecDel(n.IPAM.Type, args.StdinData); err != nil {
+			return err
 		}
-		return err
-	})
 
-	if err != nil {
-		return err
-	}
+		// There is a netns so try to clean up. Delete can be called multiple times
+		// so don't return an error if the device is already removed.
+		// If the device isn't there then don't try to clean up IP masq either.
+		var ipnets []*net.IPNet
+		err = ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
+			var err error
+			ipnets, err = ip.DelLinkByNameAddr(args.IfName)
```

Comment:
It looks like the container interface is only deleted in Layer3 mode? That doesn't seem right.

Reply:
Done